filter.h

```diff
@@ -0,0 +1,176 @@
+#ifndef _FILTER_H_
+#define _FILTER_H_
+
+//Standard Includes 
+#include <stdint.h>
+
+
+
+/** @file
+ * @defgroup MovingAvg Filter Module   
+ * 	@ingroup Filters
+ * 
+ * 
+ * This module implements a Filter using a linear transfer function that takes in scale/offset in order to calculate new values  
+ * new_value = raw_value * scale + offset 
+ * 
+ *
+ * This module defaults to a MovingAvg Filter function. The user is resposible for implementing a filter based on application. 
+ * It can be changed using the Set_Filter function 
+ * 
+ * 
+ * 
+ * 
+ * 
+ * Example for implementation with ADC module -- in progress 
+ * @code
+ * 
+ * 
+ * ADC_Init();
+ * Filter_Init();
+ * 
+ * QueueMeasurement()
+ * 
+ * 
+ * 
+ * 
+ * 
+ * 
+ * 
+ * 
+ * 
+ * @endcode
+ * 
+ * 
+ * 
+ * @author Neil Surti
+ * @author Alexander Revolous 
+ * 
+ * 
+ * **/ 
+
+
+
+
+
+
+
+
+/**
+ *  Filter_Init initliazes all variables/arrays for the filter i.e (min,max,index,) including setting the call back functions to NULL 
+ *  sets the scale/offset values from user and calculates the value for the shift  
+ * 	Sets the default filter to MovingAvgFilter
+ * 
+ * @param scale - Value that multiples with the input 
+ * @param offset - A constant value added to the linear transfer function to compute the desired result if needed
+ * 
+**/
+
+
+void Filter_Init(filter_t *, int16_t  offset, float scale); 
+
+
+
+
+/**
+ * 
+ * Filter update will be called when raw value goes through linear transfer function 
+ * new_value = (raw*scale >> shift) + offset 
+ * Calls the filter function which returns a int16_t new_value
+ * Depending on the thresholds and if min/max callbacks are set 
+ * It will call the min/max callback functions with the new filtered value
+ * 
+ * @param raw_value - Values from ADC module that will go through the linear transfer function 
+ * 
+**/
+
+void Filter_Update(uint16_t raw_value, filter_t * filter_ptr);
+
+
+
+/**
+ * 
+ *  Filter_Get will retrieve the current value from the filter instance 
+ *  
+ * @return int16_t value stored in filter_t->value 
+ * 
+ * 
+ * 
+ * 
+
+ 
+
+
+**/
+
+
+int16_t Filter_Get(filter_t *); // return value might need to be float
+
+
+
+
+
+/**
+ * 
+ * Filter_SetMin will set the MIN value in the filter to the threshold specified 
```

Comment:
The documentation is more talking about what the implementation does. The documentation in the header file should be written for the user of the module. For example instead of saying "Sets the callback function that is passed by the user" you would say "Calls the callback function with the current value when the value drops below the threshold" .